website_rma/controllers/main.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2020 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import json
+from odoo import http
+from odoo.addons.website_form.controllers.main import WebsiteForm
+from odoo.http import request
+
+
+class WebsiteForm(WebsiteForm):
+
+    def insert_record(self, request, model, values, custom, meta=None):
+        if model.model == 'rma':
+            values['partner_id'] = request.env.user.partner_id.id
+        return super(WebsiteForm, self).insert_record(
+            request, model, values, custom, meta=meta)
+
+
+class WebsiteRMA(http.Controller):
+
+    def _get_website_rma_product_domain(self, q):
+        delivered_products = request.env["stock.move"].search([
```

Comment:
You should also add a sudo in this search to avoid this ACL error:
```
{AccessError}('Sorry, you are not allowed to access this document. Only users with the following access level are currently allowed to do that:\n- Sales/Manager\n\t- Sales/User: Own Documents Only\n\t- Accounting & Finance/Billing\n\t- Inventory/User\n\t- Inventory/Manager\n\n(Document model: stock.move)', None)
```

Reply:
In fact this domain is not shouldn't be built this way, as you can request an RMA for an operation not registered in Odoo (old things before switching ERP), so I don't think we should restrict this.